app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt

```diff
@@ -1,26 +1,84 @@
         super.onCreate(savedInstanceState)
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(binding.root)
-
-        binding.button0.setOnClickListener{ binding.textView.text = "0"}
-        binding.button1.setOnClickListener{ binding.textView.text = "1"}
-        binding.button2.setOnClickListener{ binding.textView.text = "2"}
-        binding.button3.setOnClickListener{ binding.textView.text = "3"}
-        binding.button4.setOnClickListener{ binding.textView.text = "4"}
-        binding.button5.setOnClickListener{ binding.textView.text = "5"}
-        binding.button6.setOnClickListener{ binding.textView.text = "6"}
-        binding.button7.setOnClickListener{ binding.textView.text = "7"}
-        binding.button8.setOnClickListener{ binding.textView.text = "8"}
-        binding.button9.setOnClickListener{ binding.textView.text = "9"}
+        ActivityMainBinding.inflate(layoutInflater).apply {
+            resultTextView = textView
+            setUpOperandButton(
+                button0,
+                button1,
+                button2,
+                button3,
+                button4,
+                button5,
+                button6,
+                button7,
+                button8,
+                button9
+            )
+            setUpOperatorButton(
+                buttonPlus,
+                buttonMinus,
+                buttonMultiply,
+                buttonDivide
+            )
+            buttonDelete.setOnClickListener { onClickDelete() }
+            buttonEquals.setOnClickListener { onClickResult() }
+        }.also { setContentView(it.root) }
+    }
+
+    private fun setUpOperandButton(vararg operandButtons: Button) {
+        operandButtons.forEach { button ->
+            button.setOnClickListener { onClickOperand(button.text.toString()) }
+        }
+    }
+
+    private fun onClickOperand(value: String) {
+        val operand = Operand(value)
+        val currentExpression = expression.addOperand(operand)
+        refreshResultText(currentExpression)
+    }
+
+    private fun setUpOperatorButton(vararg operatorButtons: Button) {
+        operatorButtons.forEach { button ->
+            button.setOnClickListener { onClickOperator(button.text.toString()) }
+        }
+    }
+
+    private fun onClickOperator(symbol: String) {
+        val operator = Operator.findOperatorBySymbol(symbol)
+        val currentExpression = expression.addOperator(operator)
+        refreshResultText(currentExpression)
+    }
+
+    private fun onClickDelete() {
+        val currentExpression = expression.removeLast()
+        refreshResultText(currentExpression)
+    }
+
+    private fun onClickResult() {
+        Calculator.runCatching { calculate(expression.getStringExpression()) }
+            .onSuccess { result ->
+                refreshResultText(result.toString())
+            }.onFailure { Toast.makeText(this, "완성되지 않은 수식입니다.", Toast.LENGTH_SHORT).show() }
+    }
```

Comment:
```suggestion
    private fun onResultClick() {
        val expression = expression.getStringExpression()
        runCatching { Calculator.calculate(expression) }
            .onSuccess { result -> refreshResultText(result.toString()) }
            .onFailure { Toast.makeText(this, "완성되지 않은 수식입니다.", Toast.LENGTH_SHORT).show() }
    }
```
이런 식으로 작성해보는 것은 어떨까요?